tests/e2e/support.py

```diff
@@ -217,3 +236,138 @@ def delete_kueue_resources(self):
         print(f"'{self.resource_flavor}' resource-flavor deleted")
     except Exception as e:
         print(f"\nError deleting resource-flavor '{self.resource_flavor}' : {e}")
+
+
+def download_all_from_storage_bucket(
+    endpoint, access_key, secret_key, bucket_name, dir_path=dir_path
+):
+    client = Minio(
+        endpoint, access_key=access_key, secret_key=secret_key, cert_check=False
+    )
+
+    # for bucket in client.list_buckets():
+    if not os.path.exists(dir_path):
+        os.makedirs(dir_path)
+    else:
+        print(f"Directory '{dir_path}' already exists")
+    # download all files from storage bucket recursively
+    for item in client.list_objects(bucket_name, recursive=True):
+        dataset_file_path = os.path.join(dir_path, item.object_name)
+        if not os.path.exists(dataset_file_path):
+            client.fget_object(bucket_name, item.object_name, dataset_file_path)
+        else:
+            print(f"File-path '{dataset_file_path}' already exists")
+
+
+# For example : Can be used for minio storage instance deployment using minio-deployment.yaml file
+def apply_yaml_deployment(yaml_file_path):
```

Comment:
IMHO this functionality is not needed, user can setup Minio by applying yaml file directly.
Minio setup is not part of test execution, but rather environment setup.
WDYT?

Reply:
+1 Installing MinIO seems better performed in the environment setup.

Reply:
Yes right!
Actually I was intending to make this as a part of test execution but yes it can be handled in environment setup steps :eyes: 
Thanks @sutaakar @astefanutti for pointing this out and reviewing it!